File: src/main/java/com/github/kindrat/cassandra/client/filter/condition/OperatorCondition.java
package com.github.kindrat.cassandra.client.filter.condition;

import java.util.Optional;
import java.util.Set;

import static com.github.kindrat.cassandra.client.util.StringUtil.lastWord;

public class OperatorCondition implements StateCondition {
    @Override
    public boolean isCurrentState(String[] words, Set<String> columnNames) {
        Optional<String> lastWord = lastWord(words);
        return lastWord.isPresent() && columnNames.stream().anyMatch(name -> name.equalsIgnoreCase(lastWord.get()));
    }

    @Override
    public State name() {
        return State.OPERATOR;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/condition/OperatorCondition.kt
package com.github.kindrat.cassandra.client.filter.condition

import com.github.kindrat.cassandra.client.util.StringUtil

class OperatorCondition : StateCondition {

    override fun isCurrentState(words: Array<String>, columnNames: Set<String>): Boolean {
        val lastWord = StringUtil.lastWord(words)
        return lastWord?.let { word: String -> columnNames.any { it.equals(word, ignoreCase = true) }} ?: false
    }

    override fun name(): StateCondition.State {
        return StateCondition.State.OPERATOR
    }
}